### DISCUSSION TITLE
Error while using IDAKLUSolver

### DISCUSSION THREAD

**Discussion by leigedove** [pybamm-team/PyBaMM#4006] [Q&A] [answered]
Casadisolver is not good at controlling customed termination, so I tried IDAKLU solver. But there is always an error:

RuntimeError: .../casadi/core/casadi_os.cpp:166: Assertion "handle!=nullptr" failed:
PluginInterface::load_plugin: Cannot load shared library 'libcasadi_interpolant_linear.so': 
   (
    Searched directories: 1. casadipath from GlobalOptions
                          2. CASADIPATH env var
                          3. PATH env var (Windows)
                          4. LD_LIBRARY_PATH env var (Linux)
                          5. DYLD_LIBRARY_PATH env var (osx)
    A library may be 'not found' even if the file exists:
          * library is not compatible (different compiler/bitness)
          * the dependencies are not found
   )
  Tried '' :
    Error code: libcasadi_interpolant_linear.so: cannot open shared object file: No such file or directory
  Tried '.' :
    Error code: ./libcasadi_interpolant_linear.so: cannot open shared object file: No such file or directory
    
Thanks!!!

**Comment by leigedove** [comment]
The code is:

```
import pybamm
import numpy as np

options = {"thermal": "lumped"}
model = pybamm.lithium_ion.DFN(options=options)  
solver = pybamm.IDAKLUSolver()  

param = pybamm.ParameterValues("Chen2020")

def tem_cutoff(variables):
    return 310.15 - variables["X-averaged cell temperature [K]"]

tem_termination = pybamm.step.CustomTermination(
   name="Temperature cutoff", event_function=tem_cutoff)

terminations = [tem_termination, "4.2V"]

t = np.linspace(0, 20, 600)
sin_t = -15 * np.sin(0.5 * np.pi * t)
custom_current = np.column_stack([t, sin_t])

step1 = pybamm.step.current(value=custom_current, termination = terminations)

experiment = pybamm.Experiment(
    [
            step1,
            "Hold at 4.2V until 50mA",
    ]
)

init_soc = 0.001
sim = pybamm.Simulation(model, parameter_values=param, experiment=experiment, solver=solver)
sim.solve(initial_soc=init_soc)
```

**Comment by agriyakhetarpal** [comment, marked as answer]
Hi, @leigedove – this looks like a similar error that we have been receiving on in #3783. The good thing is that there is a workaround available (mentioned in the linked thread but I'll mention it here too) where you can copy the relevant missing `.so` files to the `pybamm/solvers/` folder or the better way is to set the `CASADIPATH` environment variable to where you have CasADi installed in your site-packages directory (which could be in a virtual environment or your conda/global environment).

**Reply by leigedove** [reply]
Thanks a lot! It works now
